File: packages/contracts/go/station.go
// Kitchen station contracts — added at 0.3.0 (ADR-014, Milestone 2).
// Mirrors src/types/station.ts.
//
// CONFIG aggregates under §50.1: cloud-owned, synced cloud→edge versioned by
// config_version, replaced wholesale at the edge rather than merged. The live
// ticket at a station is a Kot, which is edge-authoritative — the same
// config/operational split ADR-011 drew between RestaurantTable and
// TableSession.
//
// Field names match sqlite/0005_m2_kitchen_stations_printers.sql and
// postgres/0006_m2_kitchen_stations_printers.sql exactly.
package contracts

type Station struct {
	ID       string `json:"id"`
	OutletID string `json:"outlet_id"`
	// Stable machine code (MAIN_KITCHEN, TANDOOR, BAR, ...), unique per outlet
	// and never globally. Kot.Station stores this code, so renaming Name never
	// orphans a ticket.
	Code          string `json:"code"`
	Name          string `json:"name"`
	SortOrder     int    `json:"sort_order"`
	IsActive      bool   `json:"is_active"`
	ConfigVersion int    `json:"config_version"`
	SchemaVersion int    `json:"schema_version"`
}

// MenuItemStation is a join row, not a station_id column on MenuItem: an item
// may route to more than one station (docs/spec/kitchen.md §Stations), and both
// tickets must print.
type MenuItemStation struct {
	MenuItemID    string `json:"menu_item_id"`
	StationID     string `json:"station_id"`
	ConfigVersion int    `json:"config_version"`
	SchemaVersion int    `json:"schema_version"`
}

File: packages/contracts/src/types/station.ts
// Kitchen station contracts — added at 0.3.0 (ADR-014, Milestone 2).
//
// CONFIG aggregates under §50.1: the cloud owns them, they sync cloud→edge
// versioned by config_version, and the edge replaces them wholesale rather
// than merging. A station is a production destination (docs/spec/kitchen.md
// §Stations); it is never edge-authoritative, because which stations a kitchen
// has is a management decision, not a shop-floor transaction.
//
// The live ticket at a station is a `kot`, which IS edge-authoritative. That
// split is the same one ADR-011 drew between restaurant_table (config) and
// table_session (operational): no row is half-config, half-transaction.
//
// Field names and types match sqlite/0005_m2_kitchen_stations_printers.sql and
// postgres/0006_m2_kitchen_stations_printers.sql exactly.

import { z } from "zod";

export const StationSchema = z.object({
  id: z.string().uuid(),
  outlet_id: z.string().uuid(),
  // Stable machine code (MAIN_KITCHEN, TANDOOR, BAR, ...). Unique per outlet,
  // never globally — two outlets both having a TANDOOR is the normal case.
  // kot.station stores this code, so renaming `name` never orphans a ticket.
  code: z.string().min(1),
  name: z.string().min(1),
  sort_order: z.number().int(),
  is_active: z.boolean(),
  config_version: z.number().int(),
  schema_version: z.literal(1),
});
export type Station = z.infer<typeof StationSchema>;

// An item may route to more than one station (docs/spec/kitchen.md §Stations) —
// a thali hits MAIN_KITCHEN and TANDOOR, and both tickets must print. So this
// is a join row, not a `station_id` column on menu_item.
export const MenuItemStationSchema = z.object({
  menu_item_id: z.string().uuid(),
  station_id: z.string().uuid(),
  config_version: z.number().int(),
  schema_version: z.literal(1),
});
export type MenuItemStation = z.infer<typeof MenuItemStationSchema>;
